chore(recipe):Add a recipe for using FontAwesome (#21275)
gatsbyjs/gatsby: docs/docs/recipes/styling-css.md

- [Typography.js](/docs/typography-js/) - Another option for using Google fonts on a Gatsby site
- [The Typefaces Project Docs](https://github.com/KyleAMathews/typefaces/blob/master/README.md)
- [Live example on Kyle Mathews' blog](https://www.bricolage.io/typefaces-easiest-way-to-self-host-fonts/)

## Using FontAwesome

> Using [FontAwesome](https://fontawesome.com/) gives you access to thousands of icons for use on your site. Since Gatsby sites are React sites, we recommend you use the [react-fontawesome](https://github.com/FortAwesome/react-fontawesome) SVG library.

Reviewer: ```suggestion
Using [FontAwesome](https://fontawesome.com/) gives you access to thousands of icons for use on your site. Since Gatsby sites are React sites, it's recommended to use the [react-fontawesome](https://github.com/FortAwesome/react-fontawesome) SVG library.
```

### Prerequisites

- A [Gatsby site](/docs/quick-start)

> - The [Gatsby CLI](/docs/gatsby-cli/) installed

Reviewer: I feel like the order of these should be flipped, since you need the Gatsby CLI to build a Gatsby site. (looks like this is listed in that order on other recipes too, which should be updated..but that could be a separate PR)

### Directions

> 1. Install the react-fontawesome dependencies.

Reviewer: ```suggestion
1. Install the `react-fontawesome` dependencies.
```

```shell
npm install @fortawesome/fontawesome-svg-core  @fortawesome/free-brands-svg-icons @fortawesome/react-fontawesome
```

> > Note that there are multiple icon librarys within `react-fontawesome`. You may also be interested in `free-regular-svg-icons` and `free-solid-svg-icons` which you would install the same way.

Reviewer: ```suggestion
> Note that there are multiple icon libraries within `react-fontawesome`. You may also be interested in `free-regular-svg-icons` and `free-solid-svg-icons` which you would install the same way.
```

> 2. Import the `FontAwesomeIcon` component and the icon you want to use. Then use the icon as a component directly in your JSX.

Reviewer: ```suggestion
2. Import the `FontAwesomeIcon` component and the icon you want to use. Then use the icon as a component directly in your JSX files:
```

```jsx:title=index.js
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { faReact } from "@fortawesome/free-brands-svg-icons"

const IndexPage = () => (
  <Layout>
    <FontAwesomeIcon icon={faReact} /> //highlight-line
  </Layout>
)

export default IndexPage
```

> > This example imports a specific icon and uses it for improved performance. As an alternative you can [import the icons and build a library](https://github.com/FortAwesome/react-fontawesome#build-a-library-to-reference-icons-throughout-your-app-more-conveniently).

Reviewer: ```suggestion
> This example imports a single, specific icon and uses it for improved performance. As an alternative, you can [import the icons and build a library](https://github.com/FortAwesome/react-fontawesome#build-a-library-to-reference-icons-throughout-your-app-more-conveniently).
```
